<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Tratamento de Erros (Try, catch, throw e finally)</title>

    <script>


        // Exemplo Netflix
        var video = Array()

        video[1] = Array()
        video[1]['nome'] = 'Fullmetal Alchemist'
        video[1]['categoria'] = 'Anime'

        function getVideo(video)
        {
            try{
                console.log(video[0]['nome']) // Forçando um erro
            }

            catch(erro){
                console.log(erro) // Mostrando o erro
                console.log('Agora sim podemos tratar esse erro')
                throw new Error('Houve um erro mas não se preocupe, estamos trabalhando nisso agora!!')
            }

            finally{ 
                console.log('Sempre passa por aqui (try / catch)') // Mostrará a mensagem mesmo dando erro
            }

            console.log('A aplicação não morreu')

        }

        function tratarErro(erro)
        {
            // Lógica para registrar o erro no servidor
            console.log(erro)
        }


        getVideo(video) // Chamando a função

    </script>

</head>
<body>
    
</body>
</html>